compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/array-push-effect.expect.md

```diff
@@ -45,14 +45,7 @@ function Component(props) {
   let arr;
   if ($[4] !== x || $[5] !== y) {
     arr = [];
-    let t2;
-    if ($[7] === Symbol.for("react.memo_cache_sentinel")) {
-      t2 = {};
-      $[7] = t2;
-    } else {
-      t2 = $[7];
-    }
-    arr.push(t2);
+    arr.push({});
```

Comment:
here's an example of where the additional fixpoint loop in InferMutableRanges is suboptimal - we know that the object literal isn't mutated again, but after `inferMutableLifetimes(ir, true);` (inferring stores as mutations), `arr` and `{}` get aliased together.